spec/ics-24-host-requirements/README.md

```diff
@@ -100,6 +100,16 @@ type submitDatagram = (datagram: Datagram) => void
 
 `submitDatagram` allows relayers to relay IBC datagrams directly to the host chain. Host chains MAY require that the relayer submitting the datagram has an account to pay transaction fees, signs over the datagram in a larger transaction structure, etc - `submitDatagram` MUST define any such packaging required.
 
+### Data availability
+
+For safety (e.g. exactly-once packet delivery), host chains MUST have eventual data availability, such that any key-value pairs in state can be eventually retrieved by relayers.
+
+For liveness (relaying packets, which will have a timeout), host chains MUST have partially synchronous data availability (e.g. within a wall clock or block height bound), such that any key-value pairs in state can be retrieved by relayers within the bound.
+
+Data computable from a subset of state and knowledge of the state machine (e.g. IBC packet data, which is not directly stored) are also assumed to be available to relayers.
```

Comment:
It feels to me like there is some notion of efficiency needed here. 

I can't think of any examples in practice that require this since full nodes are required to compute it in the case where the subset is the entire state (which by definition is a subset of state), but in the case where the relayer is restricted to a proper subset of state, efficiency seems relevant.

Reply:
Added a few words about it, not sure how to formalize more concretely though.

Interestingly, there could be models where the packet data actually *is* private, I think, generated off-chain and proved in zero-knowledge to satisfy some invariants, then submitted by
the original transaction author to the destination chain (maybe even also in zero-knowledge...)